pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -332,11 +330,14 @@ private IndexedTable getIndexedTable(DataSchema dataSchema, Collection<DataTable
             function = dataTable::getString;
             break;
           case BYTES:
-            // FIXME: support BYTES in DataTable instead of converting to string
-            function = (row, col) -> BytesUtils.toByteArray(dataTable.getString(row, col));
+            function = dataTable::getBytes;
             break;
-          default:
+          case OBJECT:
```

Comment:
this is needed for intermediate aggregated object like HLL?

Reply:
Yes